autogpt_platform/backend/schema.prisma

```diff
@@ -127,34 +246,47 @@ enum AgentExecutionStatus {
   FAILED
 }
 
+// Enum for execution trigger types
+enum ExecutionTriggerType {
+  MANUAL
+  SCHEDULE
+  WEBHOOK
+}
+
 // This model describes the execution of an AgentGraph.
-model AgentGraphExecution {
-  id        String    @id @default(uuid())
-  createdAt DateTime  @default(now())
-  updatedAt DateTime? @updatedAt
-  startedAt DateTime?
+model AgentExecution {
+  id                   String               @id @default(uuid())
+  createdAt            DateTime             @default(now())
+  updatedAt            DateTime?            @updatedAt
+  startedAt            DateTime?
+  executionTriggerType ExecutionTriggerType @default(MANUAL)
 
   executionStatus AgentExecutionStatus @default(COMPLETED)
 
-  agentGraphId      String
-  agentGraphVersion Int        @default(1)
-  AgentGraph        AgentGraph @relation(fields: [agentGraphId, agentGraphVersion], references: [id, version], onDelete: Cascade)
+  agentId      String
+  agentVersion Int    @default(1)
+  Agent        Agent  @relation(fields: [agentId, agentVersion], references: [id, version], onDelete: Cascade)
+
+  // we need to be able to associate an agent execution with an agent preset
+  agentPresetId String?
+  AgentPreset   AgentPreset? @relation(fields: [agentPresetId], references: [id])
 
   AgentNodeExecutions AgentNodeExecution[]
 
-  // Link to User model
-  userId String
-  user   User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  // This is so we can track which user executed the agent.
+  executedByUserId String
+  ExecutedByUser   User   @relation(fields: [executedByUserId], references: [id], onDelete: Cascade)
```

Comment:
is there anything that could trigger an agent that this wouldn't make sense for? 

webhook? another agent?

Reply:
It is needed for to determine who it was this agent ran for, think as we know who is using what resources, for billing or even just monitoring the system. Even a webhook triggered agent or one triggered by another agent, would need an executedByUser value. In these cases, the user that setup the thing that triggered it.